nominal/experimental/stream_v2/_write_stream.py

```diff
@@ -116,9 +122,32 @@ def enqueue_from_dict(
             timestamp: The common timestamp to use for all enqueued items.
             channel_values: A dictionary mapping channel names to their values.
         """
+        timestamp_normalized = _normalize_timestamp(timestamp)
+        current_time_ns = int(datetime.now(timezone.utc).timestamp() * 1e9)
+        enqueue_dict_timestamp_diff = (current_time_ns - timestamp_normalized) / 1e9
+
         for channel, value in channel_values.items():
             self.enqueue(channel, timestamp, value)
 
+        current_time_ns = int(datetime.now(timezone.utc).timestamp() * 1e9)
+        last_enqueue_timestamp = (current_time_ns - timestamp_normalized) / 1e9
+
+        if self._track_metrics:
```

Comment:
this value isn't dynamic so it's kinda a bummer to check the condition on every append.

also do you only check this for `enqueue_from_dict`?